taca/analysis/analysis_element.py

```diff
@@ -0,0 +1,65 @@
+"""Analysis methods for sequencing runs produced by Element instruments."""
+
+import glob
+import logging
+import os
+
+from taca.element.Element_Runs import Aviti_Run
+from taca.utils.config import CONFIG
+
+logger = logging.getLogger(__name__)
+
+
+def run_preprocessing(given_run):
+    """Run demultiplexing in all data directories.
+
+    :param str given_run: Process a particular run instead of looking for runs
+    """
+
+    def _process(run):
+        """Process a run/flowcell and transfer to analysis server.
+
+        :param taca.element.Run run: Run to be processed and transferred
+        """
+        # Check if sequencing is finished. (is the final file there and was it completed OK)
+        # if sequencing is not done
```

Comment:
Maybe start with fetching the statusdb entry? The document should be updated if the status taca assigns to the run doesn't match with the statusdb document?

Reply:
Right, so in each of the steps below we want to compare the status in statusdb with the current status TACA thinks the run has, and update the document if they differ?

Reply:
Exactly.